test/distributed/_composable/fsdp/test_fully_shard_compile.py

```diff
@@ -382,51 +403,81 @@ def input_creation_fn():
 
         return model_init_fn, input_creation_fn
 
+    def _scaled_dot_product_attention_with_graph_break(
+        self, orig_fn, fullgraph, *args, **kwargs
+    ):
+        if not fullgraph:
+            torch._dynamo.graph_break()
+        return orig_fn(*args, **kwargs)
+
+    def _mock_sdpa(self, fullgraph):
+        return mock.patch.object(
+            F,
+            "scaled_dot_product_attention",
+            functools.partial(
+                self._scaled_dot_product_attention_with_graph_break,
+                F.scaled_dot_product_attention,
+                fullgraph,
+            ),
+        )
+
     @skipIfRocm
     @skip_if_lt_x_gpu(2)
-    def test_transformer_fullgraph_backend_aot_eager(self):
-        self._test_traceable_fsdp(
-            *self._create_transformer_factory_fns(), "aot_eager", fullgraph=True
-        )
+    def test_transformer_backend_aot_eager(self):
+        for fullgraph in [True, False]:
```

Comment:
nit: `@parametrize("graph_break", [True, False]`) might make debugging the test easier if e.g. the graph break path fails at some point (example: https://github.com/pytorch/pytorch/blob/main/test/autograd/test_functional.py#L684)

Reply:
The issue with distributed is that `@parametrize` directly would initialize NCCL PG for every parametrized test, which is slow. If you have what @yf225 has, it would reuse the same NCCL PG for all subtests.

I think @kwen2501 added `MultiProcContinousTest` to maybe address this. I am not sure exactly the limitations though.

Reply:
ahh